### PR TITLE
Drop Python 3.7 Support

### PR DESCRIPTION
**Pull request by vcidst**
**Proposed changes**:
- Python 3.7 is approaching its end-of-life soon, let's drop its support

**Status (please check what you already did)**:
- [ ] made PR ready for code review
- [ ] added some tests for the functionality
- [ ] updated the documentation in the [rasaHQ/rasa](https://github.com/rasaHQ/rasa)
- [ ] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa-sdk/tree/main/changelog) for instructions)
- [ ] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa-sdk#code-style) for instructions)

### CODE REVIEW

File: .github/workflows/continuous-integration.yml [hunk 1]
```diff
@@ -265,7 +265,7 @@ jobs:
 
     # deploy will only be run when there is a tag available
     if: github.event_name == 'push' && startsWith(github.event.ref, 'refs/tags')
-    needs: [quality, test, build_docker_image] # only run after all other stages succeeded
```
Comment (Tawakalt) [inline .github/workflows/continuous-integration.yml:268]: Why is this removed?

File: .github/workflows/continuous-integration.yml [hunk 2]
```diff
@@ -74,7 +74,7 @@ jobs:
     runs-on: ubuntu-22.04
     strategy:
       matrix:
-        python-version: [3.7, 3.8, 3.9, '3.10']
+        python-version: ['3.8', '3.9', '3.10']
```
Comment (Tawakalt) [inline .github/workflows/continuous-integration.yml:77]: Do they all need to be quoted?

File: .github/workflows/continuous-integration.yml [hunk 3]
```diff
@@ -211,7 +211,7 @@ jobs:
   build_docker_image:
     name: Build Docker image
     runs-on: ubuntu-22.04
-    needs: [quality, test, docker_linter, build_docker_image_set_env]
```
Comment (Tawakalt) [inline .github/workflows/continuous-integration.yml:214]: Why is this removed?

File: .github/workflows/continuous-integration.yml (hunk 1 above)
Comment (vcidst) [reply .github/workflows/continuous-integration.yml:268]: Oh, I was trying to see why doesn't the CI remove "Run Tests (3.7)" from the lists. Let me add this back at all the places

File: .github/workflows/continuous-integration.yml (hunk 2 above)
Comment (vcidst) [reply .github/workflows/continuous-integration.yml:77]: Not really, I wasn't sure about the CI. I'll revert these changes

File: changelog/975.removal.md [hunk 4]
```diff
@@ -0,0 +1 @@
+Drop Support for Python 3.7 as it is approaching its end of life
```
Comment (Tawakalt) [inline changelog/975.removal.md:1]:
```suggestion
Drop support for Python 3.7 as [it is approaching its end of life in June 2023](https://peps.python.org/pep-0537/#lifespan).
```

**Review by Tawakalt — APPROVED** [review]
